src/bandersnatch_storage_plugins/s3.py

```diff
@@ -0,0 +1,368 @@
+import configparser
+import contextlib
+import datetime
+import hashlib
+import logging
+import os
+import pathlib
+from pathlib import PurePath
+import tempfile
+from typing import (
+    IO,
+    Any,
+    Dict,
+    Generator,
+    List,
+    Optional,
+    Type,
+    Union,
+    TextIO
+)
+
+import boto3
+from botocore.client import Config
+import filelock
+from bandersnatch.storage import PATH_TYPES, StoragePlugin
+from s3path import PureS3Path, S3Path as _S3Path, register_configuration_parameter
+
+logger = logging.getLogger("bandersnatch")
+
+
+class S3Path(_S3Path):
+    def mkdir(self, mode=0o777, parents=False, exist_ok=False):
+        self.joinpath(".s3keep").touch()
```

Comment:
I guess this is to handle the bug? Is the fix released now?

Reply:
not really, https://github.com/liormizr/s3path/blob/fcee32b8034040c69b2189adfa209a1e4e33f57e/s3path.py#L286-L292 s3path can only mkdir in root dir , it cannot mkdir in s3 filesystem. As s3 does not actually support an empty folder, so an empty file must be created to keep this folder present.